src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -3259,6 +3264,22 @@ def openshift_scale(cmd, client, resource_group_name, name, compute_count, no_wa
     return sdk_no_wait(no_wait, client.create_or_update, resource_group_name, name, instance)
 
 
+def openshift_monitor_enable(cmd, client, resource_group_name, name, workspace_id, no_wait=False):
+    instance = client.get(resource_group_name, name)
+    workspace_id = _format_workspace_id(workspace_id)
+    monitor_profile = OpenShiftManagedClusterMonitorProfile(enabled=True, workspace_resource_id=workspace_id)  # pylint: disable=line-too-long
+    instance.monitor_profile = monitor_profile
+
+    return sdk_no_wait(no_wait, client.create_or_update, resource_group_name, name, instance)
+
+
+def openshift_monitor_disable(cmd, client, resource_group_name, name, no_wait=False):
+    instance = client.get(resource_group_name, name)
+    monitor_profile = OpenShiftManagedClusterMonitorProfile(enabled=False, workspace_resource_id=None)  # pylint: disable=line-too-long
+    instance.monitor_profile = monitor_profile
```

Comment:
the cluster transitioned to Non-monitored as soon as the command finished, however in the cli response the workspaceResourceId is still set to its original value. I would expect it to be null. Is there anything that I have missed or is this accepted behavior?

```
  "monitorProfile": {
    "enabled": false,
    "workspaceResourceId": "<my-workspace-id>"
  },

```

Reply:
You can add `--debug` to check the server's response. It should be null.

Reply:
confirmed with Microsoft team, this is expected behavior. Resolving this one.